### DISCUSSION TITLE
Block Material as a Function of Time

### DISCUSSION THREAD

**Discussion by jrwill11** [idaholab/moose#20902] [Q&A General] [answered]
I'm trying to simulate the transient of a moving material. The properties of the material doesn't change, but instead its block number does. 

So say I have three blocks: block 1 is LEU, and blocks 2 and 3 are air at time step 1. At time step 2, block 1 is air, block 2 is LEU, and block 3 is air. Does MOOSE have any functions where I can change the material as a function of time in a single input?

**Comment by GiudGiud** [comment, marked as answer]
Hello

The only block modifier right now is this one:
https://mooseframework.inl.gov/source/userobject/CoupledVarThresholdElementSubdomainModifier.html
You will have to encode the block change in a variable to make it work. 

Guillaume
